packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -1109,6 +1297,20 @@ describe('TransactionController', () => {
       ]);
     });
 
+    it('only reads the current chain id to filter to initially populate the metadata', async () => {
+      const getNetworkStateMock = jest.fn().mockReturnValue(MOCK_NETWORK.state);
+      const controller = newController({
+        options: { getNetworkState: getNetworkStateMock },
+      });
+
+      await controller.addTransaction({
+        from: ACCOUNT_MOCK,
+        to: ACCOUNT_MOCK,
+      });
+
+      expect(getNetworkStateMock).toHaveBeenCalledTimes(2); // we shouldn't test like this
```

Comment:
change this comment to specify where the two calls come from

Reply:
working on this

Reply:
https://github.com/MetaMask/core/pull/3643/commits/dc148dc0e15004948b45760f355ec7ce6fcb4d68